jjb/edgex-templates-java.yaml

```diff
@@ -0,0 +1,154 @@
+          branch: '{branch}'
+          submodule-recursive: '{submodule-recursive}'
+          choosing-strategy: default
+          jenkins-ssh-credential: '{jenkins-ssh-credential}'
+
+    triggers:
+      - github-pull-request:
+          trigger-phrase: '^recheck$'
+          only-trigger-phrase: false
+          status-context: '{project} Verify'
+          permit-all: true
+          github-hooks: true
+          auto-close-on-fail: false
+
+- merge_boiler_plate: &merge_boiler_plate
+    name: merge_boiler_plate
+
+    scm:
+      - lf-infra-github-scm:
+          url: '{git-clone-url}{github-org}/{project}'
+          refspec: ''
+          branch: '{branch}'
+          submodule-recursive: '{submodule-recursive}'
+          choosing-strategy: default
+          jenkins-ssh-credential: '{jenkins-ssh-credential}'
+
+    triggers:
+      - github
+      - pollscm:
+          cron: ''
+      # no reason to add github-pull-request here since it doesn't currently
+      # work for merge / push
+
+# Job Templates
+
+- job-template:
+    name: '{project-name}-{stream}-verify-java'
+
+    # Job template for Java verify jobs
+    #
+    # The purpose of this job templte is to run "maven clean install" for
+    # projects using this template.
+    #
+    # Required Variables:
+    #    branch:   git branch (eg. master)
+
+    <<: *job_boiler_plate
+    # yamllint disable-line rule:key-duplicates
+    <<: *verify_boiler_plate
+
+    builders:
+      - lf-jacoco-nojava-workaround
+      - lf-provide-maven-settings:
+          global-settings-file: 'global-settings'
+          settings-file: '{mvn-settings}'
+      - maven-target:
+          maven-version: '{mvn-version}'
+          goals: |
+              {mvn-goals}
+              -Dstream={stream}
+              {edgex-infra-mvn-opts}
+          java-opts:
+            - '{mvn-opts}'
+          settings: '{mvn-settings}'
+          settings-type: cfp
+          global-settings: 'global-settings'
+          global-settings-type: cfp
+
+- job-template:
+    name: '{project-name}-{stream}-merge-java'
+
+    # Job template for Java merge jobs
+    #
+    # The purpose of this job template is to run "maven clean install deploy"
+    # for projects using this template
+    #
+    # Required Variables:
+    #    branch:   git branch (eg. master)
+
+    <<: *job_boiler_plate
+    # yamllint disable-line rule:key-duplicates
+    <<: *merge_boiler_plate
+
+    # Defaults
+    mvn-goals: 'clean install deploy'
+
+    builders:
+      - lf-jacoco-nojava-workaround
+      - lf-provide-maven-settings:
+          global-settings-file: 'global-settings'
+          settings-file: '{mvn-settings}'
+      - maven-target:
+          maven-version: '{mvn-version}'
+          goals: |
+              {mvn-goals}
+              -Dmerge
+              -Dstream={stream}
+              {edgex-infra-mvn-opts}
+          java-opts:
+            - '{mvn-opts}'
+          properties:
+            - deployAtEnd=true
```

Comment:
Keep in mind that if any poms in hte project sets <extensions>true</extensions> this breaks.